File: BackendDotNet/Services/PublicService.cs
using BackendDotNet.DTOs.Request;
using BackendDotNet.Models;
using BackendDotNet.Repositories;

namespace BackendDotNet.Services
{
    public class PublicService : IPublicService
    {
        private readonly IContactMessageRepository _contactMessageRepository;

        public PublicService(IContactMessageRepository contactMessageRepository)
        {
            _contactMessageRepository = contactMessageRepository;
        }

        public async Task<string> SaveContactMessageAsync(ContactMessageRequestDto requestDto)
        {
            var contactMessage = new ContactMessage
            {
                Name = requestDto.Name,
                Email = requestDto.Email,
                Message = requestDto.Message
            };

            await _contactMessageRepository.SaveAsync(contactMessage);
            return "Your message has been received. Thank you!";
        }
    }
}

File: backend spring/src/main/java/com/bloodbank/service/PublicService.java
// File: src/main/java/com/bloodbank/service/PublicService.java

package com.bloodbank.service;

import com.bloodbank.dto.request.ContactMessageRequestDto;
import com.bloodbank.entity.ContactMessage;

public interface PublicService {
    ContactMessage saveContactMessage(ContactMessageRequestDto requestDto);
}